### PR TITLE
calculate coverage and post the report to coveralls

### PR DESCRIPTION
**Pull request by Songmu**
https://coveralls.io/r/mackerelio/mackerel-agent

### CODE REVIEW

File: tool/cover.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rm -f $tmpprof
+  fi
+  exit
+}
+trap cleanup INT QUIT TERM EXIT
+
+# cmd/cover doesn't support -coverprofile for multiple packages, so run tests in each dir.
+# ref. https://github.com/mattn/goveralls/issues/20
+for dir in $(find . -not -path '*/[._]*' -type d); do
+  if ls $dir/*.go > /dev/null 2>&1; then
```
Comment (motemen) [inline tool/cover.sh]: How about using `go list -f '{{.Dir}}' ./...`?

**Comment by motemen** [comment]
LGTM except for one comment

**Comment by mattn** [comment]
Low coverage rate than I had imagined.

**Comment by mattn** [comment]
Ah, tested only main.go

**Pull request merged**